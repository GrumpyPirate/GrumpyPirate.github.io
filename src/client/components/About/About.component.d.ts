import { AboutSectionFormatted } from 'store/about/about.d';

interface AboutProps {
  aboutSections: AboutSectionFormatted[];
  aboutSectionsRequest: () => void;
  hasFetched: boolean;
}

export { AboutProps };
