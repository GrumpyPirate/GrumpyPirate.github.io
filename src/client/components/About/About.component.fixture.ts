export default {
  aboutSections: [
    {
      id: 'bd147b7f-c7b3-4c83-a633-4f97d9de14fa',
      title:
        'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptate, quasi quis praesentium!',
      icons: [
        {
          name: 'Prefertile',
          large: false,
        },
        {
          name: 'Perispomenon',
          large: false,
        },
        {
          name: 'Profanably',
          large: false,
        },
        {
          name: 'Undye',
          large: true,
        },
      ],
      slug: 'ankylodontia-paralogize-zoolater',
      content:
        'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nulla id sunt __repudiandae dolorem officiis__, cumque aspernatur earum! Dolores aperiam molestias esse, possimus, quod assumenda?↵↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo a in similique voluptate nostrum sapiente minus __culpa consequatur__ quis unde! Veritatis reiciendis sapiente nihil ipsa rerum est hic omnis provident quam expedita!↵↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Molestiae ad, perferendis quam possimus quod maxime veniam corporis at vel, __facere minus eveniet__ id laboriosam doloribus, dolore accusantium atque animi. Cumque dolores aut provident, illo, nemo deserunt nam quibusdam dolorum consequuntur.↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iusto, exercitationem sunt aspernatur.',
      order: 1,
    },
  ],
  hasFetched: true,
};
