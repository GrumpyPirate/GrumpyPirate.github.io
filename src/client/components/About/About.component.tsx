import React, { useEffect, FunctionComponent } from 'react';
import { Link } from 'react-router-dom';
import ReactMarkdown from 'react-markdown';
import classnames from 'classnames';

import { AboutSectionFormatted, AboutSectionIcon } from 'types';

import Container from 'components/Layout/Container/Container.component';
import Row from 'components/Layout/Row/Row.component';
import Column from 'components/Layout/Column/Column.component';
import Page from 'components/Page/Page.component';
import PageContent from 'components/PageContent/PageContent.component';
import PageHeader from 'components/PageHeader/PageHeader.component';
import PageHeaderSubtitle from 'components/PageHeaderSubtitle/PageHeaderSubtitle.component';
import AboutSection from 'components/AboutSection/AboutSection.component';
import AboutIcon from 'components/AboutIcon/AboutIcon.component';

import classes from './About.component.scss';
import { AboutProps } from './About.component.d';

const About: FunctionComponent<AboutProps> = ({
  aboutSections,
  aboutSectionsRequest,
  hasFetched,
}: AboutProps) => {
  useEffect(() => {
    if (!hasFetched && aboutSections.length === 0) {
      aboutSectionsRequest();
    }
  }, [hasFetched, aboutSections, aboutSectionsRequest]);

  return (
    <Page>
      <PageHeader title="Hello" scrollLabel="Scroll for about" isFullscreen>
        <PageHeaderSubtitle>
          I'm a Frontend developer from Kent, working in London.
        </PageHeaderSubtitle>

        <PageHeaderSubtitle>
          <hr className={classes['about__page-header__divider']} />
          <Link to="/portfolio">Check out my portfolio</Link>
        </PageHeaderSubtitle>
      </PageHeader>

      <PageContent>
        <section className={classes['about__page-content']}>
          <Container>
            <Row justify="center">
              <Column spanSM={10} spanLG={12} spanXL={10}>
                {aboutSections.map(
                  ({ content, icons = [], id, slug, title }: AboutSectionFormatted) => (
                    <AboutSection key={`about-section--${id}`} title={title}>
                      {/* If 1 icon, house content in columns */}
                      {icons.length === 1 ? (
                        <Row justify="center">
                          <Column spanMD={8}>
                            <Row alignMD="middle">
                              {!!icons.length && (
                                <Column span={12} spanMD={4} pushMD={8}>
                                  <figure
                                    className={classnames(
                                      classes['about__section-graphic'],
                                      classes[`about__section-graphic--${slug}`],
                                      classes['about__section-graphic--single'],
                                    )}
                                  >
                                    {icons.map((icon: AboutSectionIcon) => (
                                      <AboutIcon
                                        key={`${id}__icon--${icon.name}`}
                                        iconName={icon.name}
                                        large={icon.large}
                                      />
                                    ))}
                                  </figure>
                                </Column>
                              )}
                              <Column span={12} spanMD={8} pullMD={4}>
                                <ReactMarkdown source={content} />
                              </Column>
                            </Row>
                          </Column>
                        </Row>
                      ) : (
                        <>
                          {icons.length > 0 && (
                            <figure
                              className={classnames(
                                classes['about__section-graphic'],
                                classes[`about__section-graphic--${slug}`],
                              )}
                            >
                              {icons.map((icon: AboutSectionIcon) => (
                                <AboutIcon
                                  key={`${id}__icon--${icon.name}`}
                                  iconName={icon.name}
                                  large={icon.large}
                                />
                              ))}
                            </figure>
                          )}

                          <Row justify="center">
                            <Column spanMD={8}>
                              <ReactMarkdown source={content} />
                            </Column>
                          </Row>
                        </>
                      )}
                    </AboutSection>
                  ),
                )}
              </Column>
            </Row>
          </Container>
        </section>
      </PageContent>
    </Page>
  );
};

export default About;
