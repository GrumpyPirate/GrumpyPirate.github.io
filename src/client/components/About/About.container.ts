import { connect } from 'react-redux';

import { RootState } from 'store';
import { aboutSectionsRequest } from 'store/about/about.actions';

import About from './About.component';

const mapDispatchToProps = { aboutSectionsRequest };

const mapStateToProps = (state: RootState) => ({
  aboutSections: state.about.aboutSections,
  hasFetched: state.about.hasFetched,
  error: state.about.error,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(About);
