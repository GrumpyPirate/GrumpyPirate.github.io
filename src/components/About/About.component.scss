@import 'core';

.about {
  &__page-header {
    &__divider {
      border-color: currentColor;
      border-style: solid none none;
      border-width: 1px;
      margin: 0 0 1rem;
      opacity: 0.5;

      @include respond-up('md') {
        margin-bottom: 2rem;
      }
    }
  }

  &__section-graphic {
    align-items: center;
    color: palette('theme-accent-light');
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    margin: 0 0 3rem;

    &--single {
      @include respond-up('md') {
        margin-bottom: 0;
      }
    }
  }
}
