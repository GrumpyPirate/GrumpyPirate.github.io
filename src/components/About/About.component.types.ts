import { AboutSectionFormatted } from 'types/common';

export interface AboutProps {
  aboutSections: AboutSectionFormatted[];
  aboutSectionsRequest: () => void;
  hasFetched: boolean;
}
