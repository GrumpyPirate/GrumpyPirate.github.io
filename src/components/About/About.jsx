// React
import React, { Component } from 'react'

// Routing
import { NavLink } from 'react-router-dom'

// Components
import PageContainer from 'components/PageContainer/PageContainer'
import PageContent from 'components/PageContent/PageContent'
import PageHeader from 'components/PageHeader/PageHeader'
import AboutSection from 'components/AboutSection/AboutSection'
import AboutIcon from 'components/AboutIcon/AboutIcon'

// SCSS
import './About.scss'

class About extends Component {
  render () {
    return (
      <div>
        <PageHeader title="Hello" />

        <PageContainer>
          <PageContent>
            <p>
              I'm a Frontend developer from Essex, working in London. I like coffee, video games, and cheese.
            </p>

            <p className="mb-3">
              Now that's out of the way, <NavLink to="/webdev">check out my portfolio</NavLink>.
            </p>

            {/* <AboutSection title="Languages I work with" icon="code"> */}
            <AboutSection title="Languages">
              <figure className="about__section__graphic about__section__graphic--languages">
                <AboutIcon iconName="html5" />
                <AboutIcon iconName="javascript" />
                <AboutIcon iconName="babel" />
                <AboutIcon iconName="sass" />
              </figure>

              <div className="row justify-content-center">
                <div className="col-12 col-md-8">
                  <p>
                    I care about creating accessible, quality web apps. More recently I've been working in <strong>ES6</strong> JS, using a variety of build tools. I use <strong>Babel</strong> to transpile ES6 into ES5.
                  </p>

                  <p>
                    I love <a href="http://sass-lang.com/" target="_blank">SASS</a>, but have worked with <strong>LESS</strong> and <strong>Stylus</strong>.
                  </p>
                </div>
              </div>
            </AboutSection>

            {/* <AboutSection title="Design software I use" icon="palette"> */}
            <AboutSection title="Design software">
              <figure className="about__section__graphic about__section__graphic--design">
                <AboutIcon iconName="photoshop" />
                <AboutIcon iconName="illustrator" />
                <AboutIcon iconName="sketch" />
              </figure>

              <div className="row justify-content-center">
                <div className="col-12 col-md-8 col-lg-6">
                  <p>
                    I'm comfortable using the industry standard software suites - <strong>Photoshop</strong>, <strong>Illustrator</strong> and <strong>Sketch</strong> - though given a choice, I prefer to use <strong>Illustrator</strong> when editing/exporting SVGs.
                  </p>

                  <p>
                    I've used <a href="https://zeplin.io/" target="_blank">Zeplin</a> on a few projects.
                  </p>
                </div>
              </div>
            </AboutSection>

            {/* <AboutSection title="Build tools I make apps with" icon="settingsApplications"> */}
            <AboutSection title="Build tools">
              <figure className="about__section__graphic about__section__graphic--design">
                <AboutIcon iconName="code" />
                <AboutIcon iconName="webpack" />
                <AboutIcon iconName="gulp" />
                <AboutIcon iconName="grunt" />
              </figure>

              <div className="row justify-content-center">
                <div className="col-12 col-md-8 col-lg-6">
                  <p>
                    I've worked with a variety of build tools and suites, having worked extensively with <a href="http://gulpjs.com/" target="_blank">Gulp</a>, and <a href="https://webpack.js.org/" target="_blank">Webpack</a> for React projects.
                  </p>
                </div>
              </div>
            </AboutSection>

            {/* <AboutSection title="CSS frameworks I like" icon="palette"> */}
            <AboutSection title="CSS frameworks">
              <figure className="about__section__graphic about__section__graphic--design">
                <AboutIcon iconName="bootstrap" />
                <AboutIcon iconName="foundation" />
                <AboutIcon iconName="bulma" />
              </figure>

              <div className="row justify-content-center">
                <div className="col-12 col-md-8 col-lg-6">
                  <p>
                    <strong>Bootstrap</strong> and <strong>Foundation</strong> are personal faves. I'm a huge fan of <strong>Bootstrap 4</strong>, and I incorporate it into my SASS build when given the chance. Of course, I only <code>@import</code> what's necessary!
                  </p>

                  <p>
                    Having said that, I'm confident in scaffolding my own CSS framework depending on the application needs and limitations.
                  </p>

                  <p>
                    I'm looking to give <a href="http://bulma.io/" target="_blank">Bulma</a> a go in the future.
                  </p>
                </div>
              </div>
            </AboutSection>

            {/* <AboutSection title="Application framework experience" icon="code"> */}
            <AboutSection title="App frameworks">
              <figure className="about__section__graphic about__section__graphic--design">
                <AboutIcon iconName="django" />
                <AboutIcon iconName="react" />
                <AboutIcon iconName="laravel" />
                <AboutIcon iconName="angular" />
              </figure>

              <div className="row justify-content-center">
                <div className="col-12 col-md-8 col-lg-6">
                  <p>
                    I've have a good number of <strong>Django/Django CMS</strong> projects under my belt, and have assisted development with <strong>React</strong> apps.
                  </p>

                  <p>
                    In the past, I've dabbled in <strong>Laravel (as well as Lumen)</strong>, and once made a few small <strong>Angular</strong> (v1) apps.
                  </p>
                </div>
              </div>
            </AboutSection>

            {/* <AboutSection title="Currently learning" icon="code"> */}
            <AboutSection title="Currently learning">
              <div className="row justify-content-center">
                <div className="col-12 col-md-8">
                  <div className="row align-items-md-center">
                    <div className="col-12 col-md-4 push-md-8">
                      <figure className="about__section__graphic about__section__graphic--design">
                        <AboutIcon iconName="react" largeIcon={true} />
                      </figure>
                    </div>
                    <div className="col-12 col-md-8 pull-md-4">
                      <p>
                        I'm currently learning how to develop web apps with <strong>React</strong>, and find it wonderfully refreshing. It's a hefty gear-shift, but one that makes a ton of sense.
                      </p>

                      <p>
                        I'm looking forward to creating some more complex React apps in the future.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </AboutSection>
          </PageContent>
        </PageContainer>
      </div>
    )
  } // /render ()
} // /class About extends Component

export default About
