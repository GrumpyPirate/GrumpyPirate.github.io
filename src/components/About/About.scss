/* -----------------------------------------------------------------------------
** About
** ---------------------------------------------------------------------------*/
@import 'functions';
@import 'variables';
@import 'mixins';
