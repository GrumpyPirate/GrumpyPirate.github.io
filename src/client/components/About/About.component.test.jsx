import React from 'react';
import { shallow } from 'enzyme';

import About from './About.component';

import fixture from './About.component.fixture';

describe('Components', () => {
  describe('About', () => {
    const props = {
      ...fixture,
      aboutSectionsRequest: jest.fn(),
    };

    it('should be defined', () => {
      expect(About).toBeDefined();
    });

    it('should be a function', () => {
      expect(typeof About).toBe('function');
    });

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<About {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
