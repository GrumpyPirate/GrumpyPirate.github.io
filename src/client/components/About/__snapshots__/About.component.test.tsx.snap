// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components About should render as expected, without crashing 1`] = `
<Page>
  <PageHeader
    isFullscreen={true}
    scrollLabel="Scroll for about"
    title="Hello"
  >
    <PageHeaderSubtitle>
      I'm a Frontend developer from Kent, working in London.
    </PageHeaderSubtitle>
    <PageHeaderSubtitle>
      <hr
        className="about__page-header__divider"
      />
      <Link
        to="/portfolio"
      >
        Check out my portfolio
      </Link>
    </PageHeaderSubtitle>
  </PageHeader>
  <PageContent>
    <section
      className="about__page-content"
    >
      <Container>
        <Row
          justify="center"
        >
          <Column
            spanLG={12}
            spanSM={10}
            spanXL={10}
          >
            <AboutSection
              key="about-section--bd147b7f-c7b3-4c83-a633-4f97d9de14fa"
              title="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptate, quasi quis praesentium!"
            >
              <figure
                className="about__section-graphic about__section-graphic--ankylodontia-paralogize-zoolater"
              >
                <AboutIcon
                  iconName="Prefertile"
                  key="bd147b7f-c7b3-4c83-a633-4f97d9de14fa__icon--Prefertile"
                  large={false}
                />
                <AboutIcon
                  iconName="Perispomenon"
                  key="bd147b7f-c7b3-4c83-a633-4f97d9de14fa__icon--Perispomenon"
                  large={false}
                />
                <AboutIcon
                  iconName="Profanably"
                  key="bd147b7f-c7b3-4c83-a633-4f97d9de14fa__icon--Profanably"
                  large={false}
                />
                <AboutIcon
                  iconName="Undye"
                  key="bd147b7f-c7b3-4c83-a633-4f97d9de14fa__icon--Undye"
                  large={true}
                />
              </figure>
              <Row
                justify="center"
              >
                <Column
                  spanMD={8}
                >
                  <ReactMarkdown
                    astPlugins={Array []}
                    escapeHtml={true}
                    parserOptions={Object {}}
                    plugins={Array []}
                    rawSourcePos={false}
                    renderers={Object {}}
                    skipHtml={false}
                    source="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nulla id sunt __repudiandae dolorem officiis__, cumque aspernatur earum! Dolores aperiam molestias esse, possimus, quod assumenda?↵↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo a in similique voluptate nostrum sapiente minus __culpa consequatur__ quis unde! Veritatis reiciendis sapiente nihil ipsa rerum est hic omnis provident quam expedita!↵↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Molestiae ad, perferendis quam possimus quod maxime veniam corporis at vel, __facere minus eveniet__ id laboriosam doloribus, dolore accusantium atque animi. Cumque dolores aut provident, illo, nemo deserunt nam quibusdam dolorum consequuntur.↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iusto, exercitationem sunt aspernatur."
                    sourcePos={false}
                    transformLinkUri={[Function]}
                  />
                </Column>
              </Row>
            </AboutSection>
          </Column>
        </Row>
      </Container>
    </section>
  </PageContent>
</Page>
`;
